include/caffe/datum_DB_factory.hpp

```diff
@@ -0,0 +1,98 @@
+#ifndef CAFFE_DATUMDB_FACTORY_H_
+#define CAFFE_DATUMDB_FACTORY_H_
+
+#include <map>
+#include <string>
+
+#include "caffe/common.hpp"
+#include "caffe/proto/caffe.pb.h"
+
+namespace caffe {
+
+class DatumDB;
+
+class DatumDBRegistry {
```

Comment:
If we're going to keep with the registry pattern, I wonder if we could template it away? That might get overly abstract though...